pep-0394.txt

```diff
@@ -5,104 +5,181 @@ Last-Modified: $Date$
-Python interpreter (i.e. the version invoked by the ``python`` command).
+be portable across ``*nix`` systems, regardless of the version of the Python
+interpreter invoked by the ``python`` command.
 
 * ``python2`` will refer to some version of Python 2.x.
 * ``python3`` will refer to some version of Python 3.x.
-* for the time being, all distributions *should* ensure that ``python``,
-  if installed, refers to the same target as ``python2``, unless the user
-  deliberately overrides this or a virtual environment is active.
-* however, end users should be aware that ``python`` refers to ``python3``
-  on at least Arch Linux (that change is what prompted the creation of this
-  PEP), so ``python`` should be used in the shebang line only for scripts
-  that are source compatible with both Python 2 and 3.
-* in preparation for an eventual change in the default version of Python,
-  Python 2 only scripts should either be updated to be source compatible
-  with Python 3 or else to use ``python2`` in the shebang line.
-
+* Distributions can choose whether ``python`` is installed and, if it is,
+  whether it refers to the same target as ``python2`` or ``python3``.
+* End users should be aware that the ``python`` command is inconsistent
+  across Unix-like systems.
+* In activated virtual environments, ``python`` always refers to the
+  environment's Python version.
+* Cross-platform scripts should prefer the ``python3`` shebang even if they
+  support both Python 3 and Python 2.
+* Cross-platform scripts only supporting Python 2 should use the ``python2``
+  shebang.
 
 Recommendation
 ==============
 
-* Unix-like software distributions (including systems like Mac OS X and
+For distributors
+----------------
+
+* Unix-like software distributions (including systems like macOS and
   Cygwin) should install the ``python2`` command into the default path
   whenever a version of the Python 2 interpreter is installed, and the same
   for ``python3`` and the Python 3 interpreter.
 * When  invoked, ``python2`` should run some version of the Python 2
   interpreter, and ``python3`` should run some version of the Python 3
   interpreter.
-* If the ``python`` command is installed, it should invoke the same version of
-  Python as the ``python2`` command (however, note that some distributions
-  have already chosen to have ``python`` implement the ``python3``
-  command; see the `Rationale`_ and `Migration Notes`_ below).
-* The Python 2.x ``idle``, ``pydoc``, and ``python-config`` commands should
-  likewise be available as ``idle2``, ``pydoc2``, and ``python2-config``,
-  with the original commands invoking these versions by default, but possibly
-  invoking the Python 3.x versions instead if configured to do so by the
+* If the ``python`` command is installed, it should either invoke the same
+  version of Python as the ``python3`` command or as the ``python2``
+  command.
+* Distributors may choose to set the behavior of the ``python`` command,
```

Comment:
```suggestion
* Distributors may choose to set the behavior of the ``python`` command as follows:
    - ``python2``,
    - ``python3``,
    - not provide ``python`` command
    - allow ``python`` to be configurable by an end user or a system administrator.
```

Reply:
(Note for self: several lines bellow need to be removed if this goes in).